exercises/webapp/src/components/HackerNews.vue

```diff
@@ -1,33 +1,42 @@
 <template>
   <div class="hacker-news">
     <h1>News List</h1>
-    <div v-for="item in sortedItems" :key="item.id">
-      <NewsItem v-bind:news-item="item" v-on:news_remove="onNewsRemove" v-on:update="onVoteChange"/>
+    <div v-if="sortedItems.length">
+      <div v-for="item in sortedItems" :key="item.id">
+        <NewsItem v-bind:news-item="item" v-on:news-remove="onNewsRemove" v-on:update="onVoteChange"/>
+      </div>
     </div>
-    <NewsForm v-on:news_add="onNewsAdd"></NewsForm>
+    <div v-else id="news-placeholder">
+      The list is empty :(
+    </div>
+    <NewsForm v-on:news-add="onNewsAdd" v-on:switch="onSwitch"></NewsForm>
   </div>
 </template>
 
 <script>
-import NewsItem from './NewsItem.vue';
-import NewsForm from './NewsForm.vue';
+import NewsItem from "./NewsItem.vue";
+import NewsForm from "./NewsForm.vue";
 
 export default {
-  name: 'HackerNews',
+  name: "HackerNews",
   components: {NewsItem, NewsForm},
+  props: ["initialNews"],
```

Comment:
```suggestion
  props: {
    initialNews: { type: Array, required: true }
  }
```
Gives you nice validation errors when you forget to pass down properties.